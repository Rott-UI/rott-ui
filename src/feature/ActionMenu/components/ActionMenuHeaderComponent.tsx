// React Imports
import React, {FC} from 'react'

// Constant Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {Item, Label} from '../..'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={COLOURS.GREY100}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
