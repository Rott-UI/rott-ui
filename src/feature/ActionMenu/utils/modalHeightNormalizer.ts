import {display} from '../../../utils'

export const modalHeightNormalizer = (
  dataLength: number,
  titleExist: boolean,
  subTitleExist: boolean,
  separatorTotalHeight: number,
  maxItem: number,
  itemHeight: number
) => {
  const showItemCount = dataLength > maxItem ? maxItem : dataLength
  const headerHeight = display.px(56)
  const listBottomPosition = display.px(93)

  return (
    ((showItemCount * itemHeight +
      (titleExist || subTitleExist ? headerHeight : 0) +
      listBottomPosition) *
      100) /
      display.percentage(100, 'height') +
    separatorTotalHeight
  )
}
