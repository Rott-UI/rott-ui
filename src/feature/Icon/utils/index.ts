// TODO: Icon size converter implementasyonu yapılması gerekmektedir.
