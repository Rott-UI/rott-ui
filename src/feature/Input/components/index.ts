export * from './Input'
export * from './DefaultInput'
export * from './NumericInput'
export * from './PasswordInput'
export * from './IbanInput'
export * from './PhoneInput'
export * from './CVCInput'
export * from './ExpireDateInput'
export * from './CreditCardInput'
export * from './PinPasswordInput'
export * from './DateInput'
export * from './PlateNumberInput'
export * from './AmountInput'
export * from './CheckBoxInput'
export * from './SelectInput'
export * from './InputValidation'
