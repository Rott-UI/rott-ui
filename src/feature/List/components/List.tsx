// React Imports
import {FC, forwardRef} from 'react'

// React Native Imports
import {StyleSheet} from 'react-native'

// Component Imports
import {ListStyles} from '../styles'
import {CommonUiProps, Variant} from '../../models'
import {Item} from '../../Item'

// Package Components
import {FlashList, FlashListProps} from '@shopify/flash-list'
import {Separator} from '../../Separator'

interface ListProps<T> extends CommonUiProps, FlashListProps<T> {
  renderSeparator?: boolean
  separatorWidth?: number
  separatorHeight?: number
  separatorVariant?: Variant
  headerSeparator?: boolean
  footerSeparator?: boolean
  ref?: any
}

export const List: FC<ListProps<any>> = forwardRef(
  (
    {
      renderSeparator,
      data,
      width = '100%',
      estimatedItemSize = 5,
      style,
      horizontal = false,
      separatorVariant = 'neutral-alpha-200',
      headerSeparator = false,
      footerSeparator = false,
      ...props
    },
    ref
  ) => {
    const customRenderSeparator = () => {
      return (
        <Separator
          size='full'
          orientation={!horizontal ? 'horizontal' : 'vertical'}
          variant={separatorVariant}
          width={props.separatorWidth}
          height={props.separatorHeight}
        />
      )
    }

    return (
      <Item
        style={StyleSheet.flatten([
          ListStyles({width, ...props}).defaultListContainerStyle,
          style,
        ])}>
        <FlashList
          data={data}
          ref={ref}
          estimatedItemSize={data ? (data?.length === 0 ? 1 : data.length) : estimatedItemSize}
          ItemSeparatorComponent={renderSeparator ? () => customRenderSeparator() : undefined}
          ListHeaderComponent={headerSeparator ? () => customRenderSeparator() : undefined}
          ListFooterComponent={footerSeparator ? () => customRenderSeparator() : undefined}
          horizontal={horizontal}
          {...props}
        />
      </Item>
    )
  }
)
