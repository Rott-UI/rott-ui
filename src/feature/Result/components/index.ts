export * from './Result'
export * from './ResultScreen'
