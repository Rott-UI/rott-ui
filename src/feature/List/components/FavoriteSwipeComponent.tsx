// React Imports
import {FC} from 'react'

// Component Imports
import {Icon, Item, Label, Pressable} from '../../../components'

// Constant Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {formatMessage} from '../../../utils'

interface FavoriteSwipeComponentProps {
  index: number
  favorite: boolean
  handleFavoriteChange: (itemIndex: number, itemFavorite: boolean) => void | undefined
}

export const FavoriteSwipeComponent: FC<FavoriteSwipeComponentProps> = ({
  index,
  favorite,
  handleFavoriteChange,
}) => (
  <Item row width={48}>
    <Pressable
      alignItemsCenter
      justifyContentCenter
      backgroundColor={favorite ? COLOURS.SEMANTIC_DANGER : COLOURS.SECONDARY}
      onPress={() => handleFavoriteChange && handleFavoriteChange!(index, !favorite)}>
      <Icon
        width={16}
        height={16}
        name={favorite ? 'STAR_FILL' : 'STAR'}
        variant={favorite ? 'secondary' : 'info'}
      />

      <Label textCenter color={favorite ? COLOURS.WHITE : COLOURS.BLACK} fontSize={8} marginTop={4}>
        {formatMessage(favorite ? 'LIST.ITEM.FAVORITE.REMOVE' : 'LIST.ITEM.FAVORITE.ADD')}
      </Label>
    </Pressable>
  </Item>
)
