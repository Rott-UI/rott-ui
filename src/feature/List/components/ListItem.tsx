/* eslint-disable react-native/no-inline-styles */
// React Imports
import {FC} from 'react'

// Component Imports
import {Icon, Item, Label, Pressable} from '../..'

// Util and Lib Imports
import {ListItemProps} from '../utils'
import {colourFromVariant} from '../../../utils'

// Package Imports
import Swipeable from 'react-native-gesture-handler/Swipeable'

/**
 *
 * @param icon - Solda renderlanacak icon
 * @param title - Başlık **Zorunlu**
 * @param subTitle - Alt Başlık
 * @param rightIcon - Sağda renderlanacak icon
 * @param color - Yazi Rengi
 * @param fontFamily - Yazı Fontu
 * @param onAction - Tıklandığında Alacağı Fonksiyon
 * @param backgroundColor - Item Arka Plan Rengi varsayilan degeri `green-vogue`
 * @param swipeable - Kaydirma Efekti
 * @param renderRightActions - Kaydirma Efektinde Sag Tarafta Renderlanacak Item Degerleri
 * @param renderLeftActions - Kaydirma Efektinde Sol Tarafta Renderlanacak Item Degerleri
 *
 * @returns Standart List Item Renderlar
 */
export const ListItem: FC<ListItemProps> = ({
  testID,
  icon,
  title,
  subTitle,
  rightIcon,
  titleColor = 'grey-900',
  iconColor = 'grey-900',
  rightIconColor = 'grey-900',
  fontFamily = 'Markpro-Medium',
  backgroundColor = 'grey-200',
  swipeable,
  renderRightActions,
  renderLeftActions,
  onAction,
}) => {
  const listItemContent = (
    <Pressable
      key={testID ?? `list-item-${title.toSeoFriendly()}`}
      backgroundColor={colourFromVariant(backgroundColor)}
      onPress={() => onAction && onAction!()}>
      <Item
        testID={testID ?? `list-item-${title.toSeoFriendly()}-test-id`}
        row
        alignItemsCenter
        justifyContentCenter
        height={56}
        width={390}
        paddingHorizontal={24}>
        <Item width={24}>
          {icon && (
            <Icon name={icon} width={24} height={24} variant={iconColor} noStroke mode='stroke' />
          )}
        </Item>

        <Item marginLeft={16} width={262}>
          {icon && rightIcon && (
            <Label variant={titleColor} fontSize='xl' fontFamily={fontFamily}>
              {title}
            </Label>
          )}

          {icon && !rightIcon && (
            <Label variant={titleColor} fontSize='xl' fontFamily={fontFamily}>
              {title}
            </Label>
          )}

          {!icon && !rightIcon && (
            <Label variant={titleColor} fontSize='xl' fontFamily={fontFamily} justifyContentCenter>
              {title}
            </Label>
          )}

          {subTitle && (
            <Label variant={titleColor} fontSize='lg' fontFamily={fontFamily} marginTop={8}>
              {subTitle}
            </Label>
          )}
        </Item>

        <Item width={24} marginLeft={16}>
          {rightIcon && <Icon name={rightIcon} width={24} height={24} variant={rightIconColor} />}
        </Item>
      </Item>
    </Pressable>
  )

  return (
    <>
      {swipeable && (
        <Swipeable renderRightActions={renderRightActions} renderLeftActions={renderLeftActions}>
          {listItemContent}
        </Swipeable>
      )}

      {!swipeable && {...listItemContent}}
    </>
  )
}
