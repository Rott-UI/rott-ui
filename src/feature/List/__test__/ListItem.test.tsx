// React Imports
import React from 'react'

// Component Imports
import {ListItem} from '../components'

// Util and Lib Imports
import {formatMessage, render} from '../../../utils'
import {ListItemProps} from '../utils'
import {COLOURS} from '../../../constants'
import {Pressable} from '../../Pressable/components'
import {Icon} from '../../Icon/components'
import {Label} from '../../Label/components'

describe('List Item -> Custom Component', () => {
  const dummyData: ListItemProps = {
    icon: 'SETTINGS',
    iconColor: 'white',
    title: 'TEST',
    titleColor: 'white',
    rightIcon: 'ARROW_RIGHT',
    rightIconColor: 'white',
  }

  const testId = {
    listItemTestId: `list-item-${dummyData.title.toSeoFriendly()}-test-id`,
  }

  it('list ilk render anında snapshot ile eşleşmeli', () => {
    const {icon, iconColor, title, titleColor, rightIcon, rightIconColor} = dummyData
    const rendered = render(
      <ListItem
        icon={icon}
        iconColor={iconColor}
        title={title}
        titleColor={titleColor}
        rightIcon={rightIcon}
        rightIconColor={rightIconColor}
      />
    )

    expect(rendered).toMatchSnapshot()
  })

  it('title ekranda renderlanmali', () => {
    const {title} = dummyData
    const {listItemTestId} = testId
    const {getByTestId} = render(<ListItem title={title} />)

    const listItemElement = getByTestId(listItemTestId)

    expect(listItemElement).toHaveTextContent(title)
  })

  it('title varsayilan rengi grey-900 olmali', () => {
    const {icon, title} = dummyData
    const {getByText} = render(<ListItem icon={icon} title={title} />)

    const titleElement = getByText(title)

    expect(titleElement.props.style).toHaveProperty('color', COLOURS.GREY900)
  })

  it('icon varsa ekranda renderlanmali', () => {
    const {icon, title} = dummyData
    const {listItemTestId} = testId
    const {getByTestId} = render(<ListItem icon={icon} title={title} />)

    const listItemElement = getByTestId(listItemTestId)

    // TODO: Düzenlenecek, standart oturtulacak.
    expect((listItemElement.children[0] as any).children[0].children[0].children[0]).toHaveProp(
      'name',
      icon
    )
  })

  it('iconun varsayilan rengi grey-900 olmali', () => {
    const {icon, title} = dummyData
    const {listItemTestId} = testId
    const {getByTestId} = render(<ListItem icon={icon} title={title} />)

    const listItemElement = getByTestId(listItemTestId)

    expect((listItemElement.children[0] as any).children[0].children[0].children[0]).toHaveProp(
      'variant',
      'grey-900'
    )
  })

  it('icon icin farkli bir renk verilmis ise duzgun renderlanmali', () => {
    const {icon, iconColor, title} = dummyData
    const {listItemTestId} = testId
    const {getByTestId} = render(<ListItem icon={icon} iconColor={iconColor} title={title} />)

    const listItemElement = getByTestId(listItemTestId)

    expect((listItemElement.children[0] as any).children[0].children[0].children[0]).toHaveProp(
      'variant',
      iconColor
    )
  })

  it('right icon varsa ekranda renderlanmali', () => {
    const {rightIcon, title} = dummyData
    const {listItemTestId} = testId
    const {getByTestId} = render(<ListItem rightIcon={rightIcon} title={title} />)

    const listItemElement = getByTestId(listItemTestId)

    expect(
      (listItemElement.children[listItemElement.children.length - 1] as any).children[0].children[0]
        .children[0]
    ).toHaveProp('name', rightIcon)
  })

  it('right iconun varsayilan rengi grey-900 olmali', () => {
    const {rightIcon, title} = dummyData
    const {listItemTestId} = testId
    const {getByTestId} = render(<ListItem rightIcon={rightIcon} title={title} />)

    const listItemElement = getByTestId(listItemTestId)

    expect(
      (listItemElement.children[listItemElement.children.length - 1] as any).children[0].children[0]
        .children[0]
    ).toHaveProp('variant', 'grey-900')
  })

  it('right icon icin farkli bir renk verilmis ise duzgun renderlanmali', () => {
    const {rightIcon, rightIconColor, title} = dummyData
    const {listItemTestId} = testId
    const {getByTestId} = render(
      <ListItem rightIcon={rightIcon} rightIconColor={rightIconColor} title={title} />
    )

    const listItemElement = getByTestId(listItemTestId)

    expect(
      (listItemElement.children[listItemElement.children.length - 1] as any).children[0].children[0]
        .children[0]
    ).toHaveProp('variant', rightIconColor)
  })

  it('swipe ozelliginde sag eleman var ise ekranda renderlanmali', () => {
    const {title} = dummyData
    const rightElementTestId = 'right-element-test-id'
    const {getByTestId} = render(
      <ListItem
        title={title}
        swipeable
        renderRightActions={() => (
          <Pressable
            alignItemsCenter
            justifyContentCenter
            backgroundColor={COLOURS.SECONDARY}
            testID={rightElementTestId}>
            <Icon name='STAR' width={24} height={24} />
            <Label>{formatMessage('LIST.ITEM.FAVORITE.ADD')}</Label>
          </Pressable>
        )}
      />
    )

    const rightElement = getByTestId(rightElementTestId)

    expect(rightElement).toBeOnTheScreen() // Right Element Ekranda Olmali
    expect(rightElement.children[0]).toHaveProp('name', 'STAR') // Icon Ekranda Olmali
    expect(rightElement).toHaveTextContent(formatMessage('LIST.ITEM.FAVORITE.ADD')) // Right Element Text Ekranda Olmali
  })

  it('swipe ozelliginde sol eleman var ise ekranda renderlanmali', () => {
    const {title} = dummyData
    const leftElementTestId = 'left-element-test-id'
    const {getByTestId} = render(
      <ListItem
        title={title}
        swipeable
        renderLeftActions={() => (
          <Pressable
            alignItemsCenter
            justifyContentCenter
            backgroundColor={COLOURS.SECONDARY}
            testID={leftElementTestId}>
            <Icon name='STAR' width={10} height={10} />
            <Label>{formatMessage('LIST.ITEM.FAVORITE.ADD')}</Label>
          </Pressable>
        )}
      />
    )

    const leftElement = getByTestId(leftElementTestId)

    expect(leftElement).toBeOnTheScreen() // Right Element Ekranda Olmali
    expect(leftElement.children[0]).toHaveProp('name', 'STAR') // Icon Ekranda Olmali
    expect(leftElement).toHaveTextContent(formatMessage('LIST.ITEM.FAVORITE.ADD')) // Right Element Text Ekranda Olmali
  })
})
