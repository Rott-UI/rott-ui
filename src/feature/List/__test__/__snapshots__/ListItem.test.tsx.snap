// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item -> Custom Component list ilk render anında snapshot ile eşleşmeli 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animated={false}
    backgroundColor="rgba(161, 173, 175, 1)"
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": undefined,
        "backgroundColor": "rgba(161, 173, 175, 1)",
        "bottom": undefined,
        "flex": 1,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
    testID="pressable-test-id"
  >
    <View
      alignItemsCenter={true}
      height={56}
      justifyContentCenter={true}
      paddingHorizontal={24}
      style={
        {
          "alignItems": "center",
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "row",
          "flexGrow": undefined,
          "gap": undefined,
          "height": 108,
          "justifyContent": "center",
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": 108,
          "maxWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": 46,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": 750,
          "zIndex": undefined,
        }
      }
      testID="list-item-test-test-id"
      width={390}
    >
      <View
        style={
          {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": undefined,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": 46,
            "zIndex": undefined,
          }
        }
        width={24}
      >
        <View
          style={
            {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "bottom": undefined,
              "flex": undefined,
              "flexDirection": "column",
              "flexGrow": undefined,
              "gap": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "left": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "maxHeight": undefined,
              "maxWidth": undefined,
              "overflow": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "paddingVertical": undefined,
              "position": undefined,
              "right": undefined,
              "top": undefined,
              "width": undefined,
              "zIndex": undefined,
            }
          }
        >
          <SvgMock
            fill="transparent"
            height={46}
            stroke="rgba(255, 255, 255, 1)"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={0}
            width={46}
          />
        </View>
      </View>
      <View
        marginLeft={16}
        style={
          {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": undefined,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": 31,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": 504,
            "zIndex": undefined,
          }
        }
        width={262}
      >
        <Text
          style={
            {
              "backgroundColor": undefined,
              "bottom": undefined,
              "color": "rgba(255, 255, 255, 1)",
              "flexGrow": undefined,
              "fontFamily": "Markpro-Medium",
              "fontSize": 29,
              "fontWeight": undefined,
              "gap": undefined,
              "height": undefined,
              "left": undefined,
              "letterSpacing": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "maxHeight": undefined,
              "maxWidth": undefined,
              "overflow": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "paddingVertical": undefined,
              "position": undefined,
              "right": undefined,
              "textAlign": "auto",
              "top": undefined,
              "width": undefined,
              "zIndex": undefined,
            }
          }
        >
          TEST
        </Text>
      </View>
      <View
        marginLeft={16}
        style={
          {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "bottom": undefined,
            "flex": undefined,
            "flexDirection": "column",
            "flexGrow": undefined,
            "gap": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "left": undefined,
            "marginBottom": undefined,
            "marginLeft": 31,
            "marginRight": undefined,
            "marginTop": undefined,
            "maxHeight": undefined,
            "maxWidth": undefined,
            "overflow": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "right": undefined,
            "top": undefined,
            "width": 46,
            "zIndex": undefined,
          }
        }
        width={24}
      >
        <View
          style={
            {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "bottom": undefined,
              "flex": undefined,
              "flexDirection": "column",
              "flexGrow": undefined,
              "gap": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "left": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "maxHeight": undefined,
              "maxWidth": undefined,
              "overflow": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "paddingVertical": undefined,
              "position": undefined,
              "right": undefined,
              "top": undefined,
              "width": undefined,
              "zIndex": undefined,
            }
          }
        >
          <SvgMock
            fill="transparent"
            height={46}
            stroke="rgba(255, 255, 255, 1)"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
            width={46}
          />
        </View>
      </View>
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
]
`;
